modu/source_repository/source_repository.py

```diff
@@ -86,17 +86,31 @@ def _parse_ast(self, of_file: FileInfo) -> tree_sitter.Tree:
             return ast
 
     def _discover_files(self) -> Dict[Path, FileInfo]:
+        # Get all files
+        files = [f for f in self.src_root.iterdir() if f.is_file()]
+        # Remove unwanted
+        ignored_files = self._get_ignores()
+        for ignore in ignored_files:
+            files = [n for n in files if not n.match(ignore)]
+        # Check encoding
         if self.src_root.is_file():
+            if self.src_root in ignored_files:
+                logging.warning(
+                    "Source directory is a file and is included in the .ignore file"
+                )
+                return {}
             return {self.src_root: self._inspect_file(self.src_root)}
```

Comment:
Shouldn't this check be at the beginning of `_discover_files` considering everything else we do in here isn't necessary if the root is a single file?

Reply:
It should be removed - this is deprecated from the last pr. As Chris pointed out, this is up to the user